src/net/socket_utils.zig

```diff
@@ -18,74 +19,344 @@ pub const PACKETS_PER_BATCH: usize = 64;
 // The identifier for the scoped logger used in this file.
 const LOG_SCOPE: []const u8 = "socket_utils";
 
-pub const SocketThread = struct {
-    allocator: Allocator,
-    handle: std.Thread,
+const XevThread = struct {
+    pub const Handle = std.Thread.ResetEvent;
 
-    pub fn spawnSender(
+    const List = std.DoublyLinkedList(struct {
         allocator: Allocator,
-        logger: Logger,
-        socket: UdpSocket,
-        outgoing_channel: *Channel(Packet),
-        exit: ExitCondition,
-    ) !*SocketThread {
-        return spawn(allocator, logger, socket, outgoing_channel, exit, runSender);
+        st: ?*SocketThread,
+        packet: Packet = undefined,
+        udp: xev.UDP,
+        udp_state: xev.UDP.State = undefined,
+        udp_completion: xev.Completion = undefined,
+        io: enum { idle, pending, cancelled } = .idle,
+    });
+
+    var ref_count = std.atomic.Value(usize).init(0);
+    var register = std.atomic.Value(?*List.Node).init(null);
```

Comment:
Looks like you could use std.once here? Though might not be an improvement

Reply:
Needs to track deinit + support being initialized multiple times i.e. during test/benchmark